solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java

```diff
@@ -305,16 +306,16 @@ public void init(PluginInfo info) {
             sb);
     int soTimeout =
         getParameter(args, HttpClientUtil.PROP_SO_TIMEOUT, HttpClientUtil.DEFAULT_SO_TIMEOUT, sb);
-
-    this.defaultClient =
+    this.httpClientBuilder =
         new Http2SolrClient.Builder()
             .withConnectionTimeout(connectionTimeout, TimeUnit.MILLISECONDS)
             .withIdleTimeout(soTimeout, TimeUnit.MILLISECONDS)
             .withExecutor(commExecutor)
             .withMaxConnectionsPerHost(maxConnectionsPerHost)
-            .build();
-    this.defaultClient.addListenerFactory(this.httpListenerFactory);
-    this.loadbalancer = new LBHttp2SolrClient.Builder<Http2SolrClient>(defaultClient).build();
+            .withListenerFactory(List.of(this.httpListenerFactory));
```

Comment:
I noticed an issue.  This overwrites the list of listeners that may already be present.  Shouldn't we be augmenting, not replacing?

Reply:
I was trying to work within the bounds of the existing API, but I agree this should be an add, not a replace.  I modified the API, and broke backwards-compatibility so as to not silently introduce bugs in users' applications.